compass/src/ts/compass.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {client, restAPI} from 'camelot-unchained';
+import {Promise} from 'es6-promise';
+import * as React from 'react';
+import vec2 from './vec2'
+
+export interface CompassState {
+  facing: number;
+}
+
+export interface CompassProps { }
+
+class Compass extends React.Component<CompassProps, CompassState> {
+
+  private updateSpeed: number = 100;
+
+  constructor(props: CompassProps) {
+    super(props);
+  }
+
+  angleToPercentage = (facing: number, angle: number): number => {
+    if (angle >= facing) {
+      var diff = angle - facing;
```

Comment:
let or const not var.  Normally use const, unless changing the value later, in which case use let.  Remember also variable hoisting in this particular case (the two definitions of diff are actually conflicting)